### PR TITLE
chore: upgrade angular version to rc6

### PR DESCRIPTION
**Pull request by Brocco**

### CODE REVIEW

File: addon/ng2/models/webpack-build-production.ts [hunk 1]
```diff
@@ -15,7 +15,6 @@ export const getWebpackProdConfigPartial = function(projectRoot: string, appConf
     },
     plugins: [
       new WebpackMd5Hash(),
-      new webpack.optimize.DedupePlugin(),
       new webpack.optimize.UglifyJsPlugin(<any>{
         mangle: { screw_ie8 : true, keep_fnames: true },
```
Comment (kylecordes) [inline addon/ng2/models/webpack-build-production.ts]: There was discussion in the core Angular project about a fix there that would remove the need for "keep_fnames: true" - I assume from this unchanged here, that such a fix did not make it in to RC6?

Comment (Brocco) [inline addon/ng2/models/webpack-build-production.ts]: I'm not aware of that discussion, I was just taking a crack at updating. There's still an issue with e2e tests, but I wanted to get something pushed up for others to review.

Comment (TheLarkInn) [inline addon/ng2/models/webpack-build-production.ts]: Yeah @Brocco we can remove that prop now.

Comment (Brocco) [inline addon/ng2/models/webpack-build-production.ts]: Thanks @TheLarkInn I'll strip that out and re-push, but I don't expect e2e to work based upon that change

**Comment by intellix** [comment]
With this and a newly generated project, when I serve I'm getting:

`~/@angular/compiler/src/config.js: 15 - Uncaught TypeError: Cannot read property 'Emulated' of undefined`

That line without sourcemaps contains: `__WEBPACK_IMPORTED_MODULE_0__angular_core__["a" /* ViewEncapsulation */].Emulated`

a - `undefined`
b -  `{0: "OnPush", 1: "Default", OnPush: 0, Default: 1}`

**webpack**: 2.1.0-beta.21

**Comment by antonyRoberts** [comment]
Tried reverting to webPack@2.0.0-beta and originally with 2.1.0-beta.21 and it errors out with
config.js:15 Uncaught TypeError: Cannot read property 'Emulated' of undefined

**Comment by TheLarkInn** [comment]
Whoa whoa not Webpack 2.0, TS 2.0.0. webpack version is 100% correct

**Comment by antonyRoberts** [comment]
@TheLarkInn Thanks that makes a lot more sense than reverting Webpack. alas, I'd already reverted to TS 2.0.0 and it is still  Uncaught TypeError: Cannot read property 'Emulated' of undefined,

**Comment by bjorkblom** [comment]
I got the same error, compile error on TypeScript "^2.0.0". After downgrade TypeScript to "2.0.0" the application compiles but get the ```Uncaught TypeError: Cannot read property 'Emulated' of undefined``` is shown when loading app.

**Comment by JiriBalcar** [comment]
Hello, I think, that in addon/ng2/blueprints/ng2/files/package.json should be changed typescript version to **2.0.2** because the angular rc6 depends on new **Type<T>**.

Tried that on my clone of this repo and ng serve works without any errors.

**Comment by gelliott181** [comment]
@JiriBalcar We can't require TS 2.0.2 until #1901 and #1917 are dealt with. If you're getting it to work with a 2.0.2 dependency, I'd check to see what tsc you're calling, because it's probably not the one you think you are.

**Comment by ValeryVS** [comment]
@gelliott181 #1917 was with angular rc5. #1901 probably the same.
All errors was in angular and rxjs files.
I think, typescript 2.0.2 can work with angular rc6.

**Comment by JiriBalcar** [comment]
@ValeryVS I thought the same but the errors also occurs with rc6. I think that it is because new UMD bundles. I think, that some changes in bundles loader will have to be made in angular-cli. But I do not know Webpack that well. Does anybody knows how to do that. In angular changelog is example for SystemJS.

**Comment by brandom** [comment]
I was able to get rc6 working by downgrading **webpack** to 2.1.0-beta.19 with existing CLI webpack config

**Comment by clydin** [comment]
Instead of downgrading webpack, you should be able to add
```mainFields: ['browser', 'main'],```
to the `resolve` webpack config section here: https://github.com/angular/angular-cli/blob/master/addon/ng2/models/webpack-build-common.ts#L38

This commit webpack/webpack@dc50c03 found in 2.1.0-beta.20+ seems to be the issue.  It is causing webpack to use the new es2015 bundles in rc6.

**Comment by TheLarkInn** [comment]
@clydin this doesn't make sense, RC6 is publishing es2015 bundles, shouldn't webpack be using them from the module field. I feel I'm missing something.

**Comment by Kovensky** [comment]
If it works with the regular bundle but not the es2015 bundle, then something is either wrong with the bundle, or with how the exports are declared.

For example, `export default { foo () {} }`, if naively converted to commonjs as `module.exports = { foo () {} }`, can be imported (with the current babel/webpack behavior) as `import { foo } from 'module'`, but with correct es2015 modules, it is not the case, as it is just a property in the value of the `default` export.

The modules spec appears to be moving in a direction that *will* forbid anything but default imports from commonjs modules, so even `import { foo } from 'module'` in the first example will break when eventually implementations catch up with the spec.

**Comment by keshavmesta** [comment]
I am guessing these changes weren't part of 1.0.0-beta.11-webpack.8 ??

**Pull request merged**